engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -212,6 +215,33 @@ private void standPlayer(EntityRef entity) {
         clientComp.character.send(new SetMovementModeEvent(MovementMode.WALKING));
     }
 
+    /**
+     * Auto move is disabled every time any other movement event occurs.
+     */
+    private void stopAutoMove() {
+        isAutoMove = false;
+    }
+
+    /**
+     * Start a thread that moves the entity in the camera direction at every period of 0.2 seconds.
+     * @param entity
+     */
+    private void startAutoMove(EntityRef entity) {
+        isAutoMove = true;
+        ClientComponent clientComponent = entity.getComponent(ClientComponent.class);
+        new Thread(() -> {
```

Comment:
@skaldarnar I looked up DelayManager but I could not find any implementation of it. What should my action Id be? And where exactly am I supposed to define the action that has to happen periodically?